cache.go

```diff
@@ -224,6 +224,12 @@ func getEncodingStructType(t reflect.Type) (*encodingStructType, error) {
 			copy(flds[i].cborName[n:], flds[i].name)
 			e.Reset()
 
+			encodeHead(e, byte(cborTypeByteString), uint64(len(flds[i].name)))
+			flds[i].cborNameByteString = make([]byte, e.Len()+len(flds[i].name))
+			n = copy(flds[i].cborNameByteString, e.Bytes())
+			copy(flds[i].cborNameByteString[n:], flds[i].name)
+			e.Reset()
```

Comment:
Happy to implement it that way if you prefer. I didn't worry about performance here since I was expecting it to hit the encoding struct type cache on repeated lookups, but I wouldn't be surprised to find your approach is also less error-prone over time since it reduces code duplication.